libs/template/config.go

```diff
@@ -143,11 +144,66 @@ func (c *config) skipPrompt(p jsonschema.Property, r *renderer) (bool, error) {
 	return true, nil
 }
 
+func (c *config) promptSelect(name, description string, propertySchema *jsonschema.Schema) error {
+	// List of options to display to user
+	options, err := propertySchema.EnumStringSlice()
+	if err != nil {
+		return err
+	}
+
+	// Get user input
+	userInput, err := cmdio.AskSelect(c.ctx, description, options)
+	if err != nil {
+		return err
+	}
+
+	// Convert user input string back to a value
+	c.values[name], err = propertySchema.ParseString(userInput)
+	if err != nil {
+		return err
+	}
+
+	// Validate the partial config based on this update
+	return c.schema.ValidateInstance(c.values)
+}
+
+func (c *config) promptText(name, description, defaultVal string, propertySchema *jsonschema.Schema) error {
+	for {
+		// Get user input. Parse the string back to a value
+		userInput, err := cmdio.Ask(c.ctx, description, defaultVal)
+		if err != nil {
+			return err
+		}
+
+		// Convert user input string back to a Go value
+		c.values[name], err = propertySchema.ParseString(userInput)
+		// Error parsing user input. Retry if parsing fails.
+		if err != nil {
+			target := &jsonschema.ParseStringError{}
+			if errors.As(err, target) {
+				cmdio.LogString(c.ctx, fmt.Sprintf("Validation failed: %q is not a %s", target.Value, target.ExpectedType))
```

Comment:
If we reformat the error here, why do we have the formatting in its `Error()` function?

As this `err != nil` block retries regardless of the type, it can defer to the underlying error for formatting.